### PR TITLE
done task 4-styling

### PR DESCRIPTION
**Pull request by demasha0908**
https://demasha0908.github.io/fe-react-2024/

- [x]  Mentor is added to pr
- [x]  AC copied to pr description
- [x]  Link to deployed version is added to pr
- [x]  Added layout of Search Bar
- [x]  Rewritten all styles with Module Styles
- [x]  Implemented dynamic theme change logic

### CODE REVIEW

File: src/components/product/ProductList.component.tsx [hunk 1]
```diff
@@ -8,11 +9,14 @@ import ProductCard from './ProductCard.component';
 import styles from './Product.module.css';
 
 const ProductList = () => (
-    <ul className={styles.product__list}>
-        {data.map((product: Product) => (
-            <ProductCard key={product.id} product={product} />
-        ))}
-    </ul>
+    <section className={styles.product__section}>
```
Comment (vladyslav-yermolin-moc) [inline src/components/product/ProductList.component.tsx:12]: Please correct border and searc text color 
image

File: src/components/searchbar/SearchBar.module.css [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+  gap: 10px;
+}
+
+.filter__button {
+  color: var(--text-secondary);
+  padding: 15px 20px;
+  font-weight: 600;
+  box-sizing: border-box;
+  border: 1px solid var(--border-secondary);
+  border-radius: 20px;
```
Comment (vladyslav-yermolin-moc) [inline src/components/searchbar/SearchBar.module.css:79]: Please correct border-radius
image

File: src/components/footer/footer.module.css
Comment (vladyslav-yermolin-moc) [inline src/components/footer/footer.module.css:1]: Please correct 
- separator color and gap to the page edge and to the page content
- Padding of the footer's text
image

File: src/App.tsx [hunk 3]
```diff
@@ -9,18 +9,25 @@ import styles from './App.module.css';
...
 
     return (
         <>
-            <HeaderComponent shouldShowAbout={shouldShowAbout} toggleAboutState={toggleAbout} />
-            <main>
-                <div className={styles.container}>{shouldShowAbout ? <AboutComponent /> : <ProductList />}</div>
-            </main>
-            <FooterComponent />
+            <div className={`${theme}-theme`}>
+                <HeaderComponent shouldShowAbout={shouldShowAbout} toggleAboutState={toggleAbout} showTheme={changeTheme} />
```
Comment (vladyslav-yermolin-moc) [inline src/App.tsx:25]:
```suggestion
                <HeaderComponent shouldShowAbout={shouldShowAbout} toggleAboutState={toggleAbout} changeTheme={changeTheme} />
```

File: src/App.tsx [hunk 4]
```diff
@@ -9,18 +9,25 @@ import styles from './App.module.css';
 
 function App() {
     const [shouldShowAbout, setShouldShowAbout] = useState<boolean>(true);
+    const [theme, setTheme] = useState('light');
+    const changeTheme = () => {
+        const stateTheme = theme === 'light' ? 'dark' : 'light';
```
Comment (vladyslav-yermolin-moc) [inline src/App.tsx:14]:
```suggestion
        const newTheme = theme === 'light' ? 'dark' : 'light';
```

**Review by vladyslav-yermolin-moc — CHANGES_REQUESTED** [review]
Thanks for a good work, @demasha0908

File: src/components/searchbar/SearchBar.module.css (hunk 2 above)
Comment (demasha0908) [reply src/components/searchbar/SearchBar.module.css:82]: fixed

File: src/components/footer/footer.module.css
Comment (demasha0908) [reply src/components/footer/footer.module.css:1]: fixed

File: src/App.tsx (hunk 3 above)
Comment (demasha0908) [reply src/App.tsx]: fixed

File: src/App.tsx (hunk 4 above)
Comment (demasha0908) [reply src/App.tsx]: fixed

**Comment by vladyslav-yermolin-moc** [comment]
Hey, @demasha0908, thanks for your fixes.
Something happened with your light theme - here is screencast with description

https://github.com/demasha0908/fe-react-2024/assets/98031768/b70ff596-3a0d-4425-8a2c-b9c1b509e23b

**Review by vladyslav-yermolin-moc — CHANGES_REQUESTED** [review]
Please see my comment.
Also pay attantion to the active state of the icons of themes: sun and moon.

**Review by vladyslav-yermolin-moc — APPROVED** [review]
Thanx for your fixes, @demasha0908 !
Please correct in the next pr 
image

**Pull request merged**